Improve cache key graph printing performance

aorenste opened the pull request:
Teach the graph printer how to allow overriding printing SymTypes (`SymInt`, `SymFloat`, `SymBool`) and then use that to reuse the fast SymNode printing from `torch._inductor.utils.sympy_str()` to make computing the cache key faster.

On my computer the repro from #151823 goes from 480s -> 80s (still terrible... but better).

Fixes #151823 

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #151928



cc @ezyang @SherlockNoMad @EikanWang @jgong5 @wenzhe-nrv @voznesenskym @penguinwu @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

aorenste committed:
Improve cache key graph printing performance

[ghstack-poisoned]

laithsakka commented on torch/_inductor/compile_fx.py:
```diff
@@ -1130,9 +1130,17 @@ def codegen_and_compile(
-                inductor_post_grad_graph_str = gm.print_readable(
-                    print_output=False, include_stride=True, include_device=True
-                )
+
```
is it possible to push this logic inside print_readable? and we can pass fast_sympy_print=True?
```
gm.print_readable(print_output=False, include_stride=True, include_device=True, fast_sympy_print=True)
```
The reason is that i expect i will have to apply this on more places than just this .
namely all structured tracing logs as well.

laithsakka reviewed (approved):
looks good just one comment about pushing the override logic inside print_readable to make it easier to replicate this on other places and reduce code repetition

aorenste committed:
Update on "Improve cache key graph printing performance"

Teach the graph printer how to allow overriding printing SymTypes (`SymInt`, `SymFloat`, `SymBool`) and then use that to reuse the fast SymNode printing from `torch._inductor.utils.sympy_str()` to make computing the cache key faster.

On my computer the repro from #151823 goes from 480s -> 80s (still terrible... but better).

Fixes #151823 




cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

aorenste commented:
test_modules_can_be_imported looks unrelated (appears in several different PRs)
@pytorchbot merge -i
